code/game/objects/structures/window.dm

```diff
@@ -460,7 +460,7 @@
 			else if (tool.tool_behaviour)
 				to_chat(user, span_warning("The bolts need to be loosened first!"))
 
-	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	return ..()
```

Comment:
I think this should cancel the attack chain if `tool.tool_behaviour` is valid, because behaviour is guaranteed from above if the tool has that variable defined.